### PR TITLE
feat: Add a warning banner for end-of-life versions

### PR DESCRIPTION
**Pull request by fhennig**
This PR adds a warning to the top of articles if a version is selected that is no longer maintained. This requires the `asciidoc.attributes.page-end-of-life` variable to be set to true in the `antora.yml` file. If it isn't set, not banner is displayed (this means that this change will not have any effect unless the variable is set).

The banner styling is done with the `unmaintained-version-banner` CSS class.

This is what the banner currently looks like: 

image

## Example antora.yml config snippet:

```yaml
asciidoc:
  attributes:
    # Whether this version is already end of life. 
    # If true, a banner will be displayed informing the user.
    end-of-life: true
    # to make attributes accessible to the UI template, they need to
    # be prefixed with "page-"
    page-end-of-life: "{end-of-life}"
```

### CODE REVIEW

File: src/css/warning-banner.css [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.unmaintained-version-banner {
```
Comment (Techassi) [inline src/css/warning-banner.css]: suggestion: Maybe use `eol-banner` or `end-of-life-banner` as the class name to keep it consistent with the page attribute.

File: src/partials/article.hbs [hunk 2]
```diff
@@ -1,4 +1,14 @@
 <article class="doc">
+{{! In antora.yml set asciidoc.attributes.page-end-of-life to true/false}}
+{{#if page.attributes.end-of-life}}
+<div class="unmaintained-version-banner">
+  <b>This version of the Stackable Data Platform is no longer maintained.</b>
```
Comment (Techassi) [inline src/partials/article.hbs]: suggestion: Make this a headline, like `h5` or `h6`.

File: src/partials/article.hbs [hunk 3]
```diff
@@ -1,4 +1,14 @@
...
+  <br>
+  {{#if page.canonicalUrl}}
+  Please read the <a href="{{{page.canonicalUrl}}}">latest version of this page</a> instead.
```
Comment (Techassi) [inline src/partials/article.hbs]: suggestion: This text should live in its own element like `p`.

**Comment by fhennig** [comment]
This is what it looks like now:

image

File: src/css/warning-banner.css (hunk 1 above)
Comment (Techassi) [reply src/css/warning-banner.css]: Resolved in 17bf0f3

File: src/partials/article.hbs (hunk 2 above)
Comment (Techassi) [reply src/partials/article.hbs]: Resolved in 17bf0f3

File: src/partials/article.hbs (hunk 3 above)
Comment (Techassi) [reply src/partials/article.hbs]: Resolved in 17bf0f3

**Review by Techassi — APPROVED** [review]

**Pull request merged**